### PR TITLE
docker: fix gitian support, optimize container layer count and improve script consistency

### PR DESCRIPTION
**Pull request by kittywhiskers**

### CODE REVIEW

File: contrib/containers/develop/Dockerfile [hunk 1]
```diff
@@ -8,19 +8,34 @@ INCLUDE+ ci/Dockerfile
...
+    zsh \
+    && \
+    rm -rf /var/lib/apt/lists/*
+
 # Discourage root access, this is an interactive instance
 #
 # Sometimes these commands are run repetitively _after_ the non-sudo
 # user was introduced and therefore these commands would fail
-# To mitigate the build halting, we've added "|| true" so that it
+# To mitigate the build halting, you can add "|| true" so that it
```
Comment (kxcd) [inline contrib/containers/develop/Dockerfile:30]: Or `;:` to return true, two characters 😂

Comment (kittywhiskers) [reply contrib/containers/develop/Dockerfile:30]: We try to aim code to be self-documenting and readability usually helps in that regard but out of curiosity, will this work with most shells or is this a bash-ism?

Comment (kxcd) [reply contrib/containers/develop/Dockerfile:30]: Yep, it is a bashism. The `;` is standard for any shell to chain next command without checking exit status of the previous, `:` generates a `0` exit code (true), while `!:` generates a `1` exit code (false).

Comment (UdjinM6) [reply contrib/containers/develop/Dockerfile:30]: Interesting but I'd rather keep it more readable/cross-compatible.

**Comment by UdjinM6** [comment]
LGTM but I'd like @strophy to review it too.

File: contrib/containers/develop/Dockerfile (hunk 1 above)
Comment (kxcd) [reply contrib/containers/develop/Dockerfile:30]: I agree that using `true` is more readable, I disagree we should still be trying to write POSIX compatible scripts in 2022, `bash` is ubiquitous in all *NIX platforms, you'll never move to a different shell.  Reasons to use the above syntax is for performance, one of the most expensive things you can do on Linux is create a new process and `true` and `false` are themselves processes, you should always prefer to use internal functions of the shell where ever possible to speed it up. That said, I'm not married to the idea, I just thought you might find it interesting. 😁

Comment (kittywhiskers) [reply contrib/containers/develop/Dockerfile:30]: It is quite interesting, I didn't consider the process spawn overhead and I'll be sure to keep this in mind. Thanks a lot for sharing this with us!

**Comment by strophy** [comment]
LGTM, I have a few changes to `Dockerfile.GitHubActions.Dispatch` that I will send in a separate PR after this is merged. I think the old `Dockerfile` based on `phusion/baseimage:bionic-1.0.0` can be deleted entirely, we don't use this base image anymore.

**Review by strophy — APPROVED** [review]

**Review by PastaPastaPasta — APPROVED** [review]
utACK for squash merge

**Review by UdjinM6 — APPROVED** [review]
utACK

**Pull request merged**